### PR TITLE
86cuex03r/UI/Custom-Confirm-Dialog

### PR DESCRIPTION
**Pull request by RJR-003**
### Overview
This pull request includes UI for a reusable Confirmation box.

### Description
- Made a reusable custom Confirmation modal with agree and disagree buttons.
- Accepts extra functions to be performed while accepting or declining by adding props.
- Made the confirmation box responsive.
- For example, this box will be used for the trip cancellation option.

### CODE REVIEW

File: src/components/ui/ConfirmDialog/ConfirmDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        setOpen(true);
+    };
+
+    const handleClose = () => {
+        setOpen(false);
+    };
+
+    return (
+        <>
+            <Button variant="contained" onClick={handleClickOpen}>
```
Comment (seraGeorge) [inline src/components/ui/ConfirmDialog/ConfirmDialog.tsx]: Move the button outside the component.

File: src/components/ui/ConfirmDialog/ConfirmDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+import useMediaQuery from '@mui/material/useMediaQuery';
+import { useTheme } from '@mui/material/styles';
+import { useState } from 'react';
+
+interface ConfirmDialogProps {
+    title?: string;
+    agreeText?: string;
```
Comment (seraGeorge) [inline src/components/ui/ConfirmDialog/ConfirmDialog.tsx]: Make this mandatory.

File: src/components/ui/ConfirmDialog/ConfirmDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            <Dialog
+                fullScreen={fullScreen}
+                open={open}
+                onClose={handleClose}
+                aria-labelledby="responsive-dialog-title"
+            >
+                <DialogTitle id="responsive-dialog-title">
+                    {props.title}
+                </DialogTitle>
+                <DialogContent>
```
Comment (ashwin8448) [inline src/components/ui/ConfirmDialog/ConfirmDialog.tsx:54]: Add close button

**Review by ashwin8448 — CHANGES_REQUESTED** [review]

**Review by seraGeorge — CHANGES_REQUESTED** [review]

**Review by seraGeorge — APPROVED** [review]

File: src/components/ui/ConfirmDialog/ConfirmDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+import CloseIcon from '@mui/icons-material/Close';
+import useMediaQuery from '@mui/material/useMediaQuery';
+import { useTheme } from '@mui/material/styles';
+import { IconButton } from '@mui/material';
+
+interface ConfirmDialogProps {
```
Comment (akshay-krishna) [inline src/components/ui/ConfirmDialog/ConfirmDialog.tsx]: If interface, better to name it as IConfirmDialogProps

**Review by akshay-krishna — APPROVED** [review]

**Review by ashwin8448 — APPROVED** [review]

File: src/components/ui/ConfirmDialog/ConfirmDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+                        <Button
+                            autoFocus
+                            onClick={() => {
+                                handleDisagreeFunction();
+                                props.handleClose();
+                            }}
+                        >
+                            {props.disagreeText}
+                        </Button>
+                    ) : null}
```
Comment (visakhvqburst) [inline src/components/ui/ConfirmDialog/ConfirmDialog.tsx]: Do we need this ternary operator here?

Comment (RJR-003) [reply src/components/ui/ConfirmDialog/ConfirmDialog.tsx]: No we dont need that, I will change and use "&&"

**Pull request merged**